tests/dragonfly/seeder_test.py

```diff
@@ -14,6 +14,18 @@ async def test_static_seeder(async_client: aioredis.Redis):
     assert abs(await async_client.dbsize() - 10_000) <= 50
 
 
+@dfly_args({"proactor_threads": 4})
+async def test_static_element_ratio(async_client: aioredis.Redis):
+    s = StaticSeeder(
+        key_target=10, data_size=10_000, variance=1, samples=1, element_size_ratio=1, types=["LIST"]
+    )
+    await s.run(async_client)
+
+    keys = await async_client.keys()
+    assert (await async_client.llen(keys[0])) == 1
+    assert len(await async_client.lpop(keys[0])) == 10_000
```

Comment:
A question:

if `element_size_ratio=1/2` then data_size is `10k ** (1/2)` == 100 and variance is 1. So `dsize=100`. My question is, why do we `xor` the `dsize` ? That is: `LG_funcs.esize = math.ceil(dsize ^ delement_ratio)` which is how many elements a given type should contain (that is `llen(keys[0])`).

So to sumamrize:

1. Why do we `xor` this ? `LG_funcs.esize = math.ceil(dsize ^ delement_ratio)`
2. Why do we express the number of elements per set via all of this? Can't we just be specific on how many elements we want of a given size each ?

There is something I am missing so I am asking here :smile:

Reply:
1. It's not xor, it's power 🙃
2. If it's too difficult and fragile, no one will use it properly. It's just a 0/1 slider: 0 means smallest possible elements, 1 means biggest possible

Reply:
For 1. power is `**` not caret `^` which is `xor` ?

Reply:
https://www.lua.org/manual/2.4/node9.html#:~:text=%2C%20/%20and-,%5E%20(exponentiation),-%2C%20and%20the%20unary

Reply:
I brainfarted. It's LUA not python :face_exhaling: :facepalm: 

Now it all makes sense. Ignore my blindness....

Reply:
Lol @dranikpg we make the comment at exact same time. I did not read your link but somehow I noticed and then you replied at the exact same moment I figured and reply :rofl: